src/codegen/compile.ml

```diff
@@ -4488,7 +4491,8 @@ module Serialization = struct
           (* Second time we see this *)
           (* Calculate relative offset *)
           let (set_offset, get_offset) = new_local env "offset" in
-          get_tag ^^ get_data_buf ^^ G.i (Binary (Wasm.Values.I32 I32Op.Sub)) ^^
+          get_x ^^ Heap.load_field Tagged.tag_field ^^
+          get_data_buf ^^ G.i (Binary (Wasm.Values.I32 I32Op.Sub)) ^^
           set_offset ^^
           (* A sanity check *)
           get_offset ^^ compile_unboxed_const 0l ^^
```

Comment:
'office' below should be `relative offset`

Reply:
https://github.com/dfinity/motoko/pull/2810